### PR TITLE
Add method to add padding to video

### PR DESCRIPTION
**Pull request by RalfNieuwenhuizen**

### CODE REVIEW

**Review by FrikkieSnyman — COMMENTED** [review]
Lekker. This is not needed for ios?

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java [hunk 1]
```diff
@@ -400,6 +400,37 @@ static void trim(ReadableMap options, final Promise promise, ReactApplicationCon
...
+    cmd.add("-vf");
+    cmd.add("scale="+width+":"+height+":force_original_aspect_ratio=decrease,pad="+width+":"+height+":(ow-iw)/2:(oh-ih)/2,setsar=1");
+
+    cmd.add("-preset");
+    cmd.add("ultrafast");
+    // NOTE: DO NOT CONVERT AUDIO TO SAVE TIME
+    cmd.add("-c:a");
+    cmd.add("copy");
+    // NOTE: FLAG TO CONVER "AAC" AUDIO CODEC
+    cmd.add("-strict");
```
Comment (jaspermeijaard) [inline android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java]: Why did you add/enable `-strict` and `-2` flags for the `pad` method? I don't know the reason but they are disabled in the `trim` method.

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java [hunk 2]
```diff
@@ -400,6 +400,37 @@ static void trim(ReadableMap options, final Promise promise, ReactApplicationCon
...
+
+    ArrayList<String> cmd = new ArrayList<String>();
+    cmd.add("-y"); // NOTE: OVERWRITE OUTPUT FILE
+
+    // NOTE: INPUT FILE
+    cmd.add("-i");
+    cmd.add(source);
+
+    cmd.add("-vf");
+    cmd.add("scale="+width+":"+height+":force_original_aspect_ratio=decrease,pad="+width+":"+height+":(ow-iw)/2:(oh-ih)/2,setsar=1");
```
Comment (jaspermeijaard) [inline android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:419]: In https://github.com/relivecc/react-native-video-processing/pull/4 we plan to update from 3 (this branch) to version 4. Make sure you check the scale parameter against that branch and/or version.

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java [hunk 3]
```diff
@@ -400,6 +400,37 @@ static void trim(ReadableMap options, final Promise promise, ReactApplicationCon
     executeFfmpegCommand(cmd, tempFile.getPath(), ctx, promise, "Trim error", null);
   }
 
+  static void pad(ReadableMap options, final Promise promise, ReactApplicationContext ctx) {
```
Comment (jaspermeijaard) [inline android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:403]: I can't find the React implementation of this method but I expect it to be standalone functionalities. But if we plan to invoke this method after/before the trim method, we might want to integrate the methods so it only runs ffmpeg for a single video once and not twice.

**Comment by RalfNieuwenhuizen** [comment]
> Lekker. This is not needed for ios?

@FrikkieSnyman Instagram does have capable iOS developers 🤷  - it scales okay there by default
IMG_5615D9AA5496-1

File: lib/ProcessingManager/ProcessingManager.android.js [hunk 4]
```diff
@@ -28,6 +30,20 @@ export class ProcessingManager {
       .then((res) => res.source);
   }
 
+  static pad(source: sourceType, options: padOptions): Promise<string> {
```
Comment (jaspermeijaard) [inline lib/ProcessingManager/ProcessingManager.android.js:33]: How did you feel after writing non-type-safe code 😂 ?

**Comment by jaspermeijaard** [comment]
👍

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java (hunk 3 above)
Comment (RalfNieuwenhuizen) [reply android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:403]: https://github.com/relivecc/skyhawk-mobileapp/pull/3386

File: lib/ProcessingManager/ProcessingManager.android.js (hunk 4 above)
Comment (RalfNieuwenhuizen) [reply lib/ProcessingManager/ProcessingManager.android.js:33]: 😕

**Comment by RalfNieuwenhuizen** [comment]
@jaspermeijaard thanks for the extensive review! I'll check these changes with the new version as well. The PR I added above should answer your other question!

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java (hunk 3 above)
Comment (jaspermeijaard) [reply android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:403]: 👍

File: android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java (hunk 2 above)
Comment (RalfNieuwenhuizen) [reply android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:419]: Works! ✅

Comment (jaspermeijaard) [reply android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java:419]: Awesome 👍 Great job
